File: SkyEditor.Core.Windows/Windows/Utilities/Json.vb
Namespace Windows.Utilities
    Partial Public Class Json
        ''' <summary>
        ''' Serializes the given object into JSON and writes it to disk.
        ''' </summary>
        ''' <param name="Filename">Filename to store the JSON.</param>
        ''' <param name="ObjectToSerialize">Object to serialize.</param>
        Public Shared Sub SerializeToFile(Filename As String, ObjectToSerialize As Object)
            SkyEditor.Core.Utilities.Json.SerializeToFile(Filename, ObjectToSerialize, New Windows.Providers.WindowsIOProvider)
        End Sub

        ''' <summary>
        ''' Deserializes JSON stored on disk.
        ''' </summary>
        ''' <typeparam name="T"></typeparam>
        ''' <param name="Filename">Path to the text file containing a JSON string.</param>
        ''' <returns></returns>
        Public Shared Function DeserializeFromFile(Of T)(Filename As String) As T
            Return SkyEditor.Core.Utilities.Json.DeserializeFromFile(Of T)(Filename, New Windows.Providers.WindowsIOProvider)
        End Function
    End Class
End Namespace

File: SkyEditor.Core/Utilities/Json.cs
using Newtonsoft.Json;
using SkyEditor.Core.IO;
using SkyEditor.IO.FileSystem;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkyEditor.Core.Utilities
{
    /// <summary>
    /// Converts objects to and from JSON
    /// </summary>
    public static class Json
    {
        /// <summary>
        /// Serializes the specified object into a JSON string.
        /// </summary>
        /// <param name="obj">Object to serialize.</param>
        /// <returns>Json text that represents the given object.</returns>
        public static string Serialize(object obj)
        {
            return JsonConvert.SerializeObject(obj);
        }

        /// <summary>
        /// 
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="json"></param>
        /// <returns></returns>
        public static T Deserialize<T>(string json)
        {
            return JsonConvert.DeserializeObject<T>(json);
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="type"></param>
        /// <param name="json"></param>
        /// <returns></returns>
        public static object Deserialize(Type type, string json)
        {
            return JsonConvert.DeserializeObject(json, type);
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="filename"></param>
        /// <param name="obj"></param>
        /// <param name="provider"></param>
        public static void SerializeToFile(string filename, object obj, IFileSystem provider)
        {
            provider.WriteAllText(filename, Serialize(obj));
        }

        /// <summary>
        /// 
        /// </summary>
        /// <typeparam name="T"></typeparam>
        /// <param name="filename"></param>
        /// <param name="provider"></param>
        /// <returns></returns>
        public static T DeserializeFromFile<T>(string filename, IFileSystem provider)
        {
            return Deserialize<T>(provider.ReadAllText(filename));
        }
    }
}
